trainee_template/synthesis_bounds.amlg

```diff
@@ -129,12 +129,163 @@
 					(append featureBoundsMap feature_bounds_map)
 				)
 		))
+
+		(assign (assoc
+			constrained_features
+				(indices (filter
+					(lambda (!= (null) (get (current_value) "constraint")))
+					feature_bounds_map
+				))
+		))
+	)
+
+	;Helper method for synthesis to verify all the constrained feature values are within their constraints
+	;verification is done in the same order as action_features and if it fails, will output a (null) for the generated case
+	#!VerifyConstrainedFeatureValues
+	(let
+		(assoc
+			generated_case_map (zip context_features context_values)
+			residuals_tuples_map (zip rand_ordered_features threshold_feature_residuals)
+		)
+
+		(map
+			(lambda (let
+				(assoc feature (current_value 1) )
+
+				(declare (assoc
+					new_feature_value (get generated_case_map feature)
+					action_value (get generated_case_map feature)
+					feature_residual (get residuals_tuples_map (list feature 0))
+				))
+				(if (get nominalsMap feature)
+					(if (call !ConstrainNominalFeature)
+						(assign (assoc constraint_function_failed (true)))
+					)
+
+					(if (call !LimitGeneratedValueToFeatureBounds)
+						(assign (assoc constraint_function_failed (true)))
+					)
```

Comment:
I've added a conclude to stop iteration early, but we still need to set the variable because for flows where new case generation is required, it's checked during uniqueness validation to fast-track the retry/regen mechanism.